dev/WebView2/WebView2.cpp

```diff
@@ -0,0 +1,1932 @@
+                m_coreWebViewController = nullptr;
+                m_coreWebView = nullptr;
+                ResetProperties();
+            }
+
+            FireCoreProcessFailedEvent(args);
+        }});
+
+    m_cursorChangedRevoker = m_coreWebViewCompositionController.CursorChanged(winrt::auto_revoke, {
+        [this](auto const& controller, auto const& obj)
+        {
+            m_requestedCursor = controller.Cursor();
+
+            UpdateCoreWindowCursor();
+        }});
+}
+
+winrt::IAsyncAction WebView2::CreateCoreObjects()
+{
+    MUX_ASSERT((m_isImplicitCreationInProgress && !m_isExplicitCreationInProgress) ||
+               (!m_isImplicitCreationInProgress && m_isExplicitCreationInProgress));
+
+    if (m_isClosed)
+    {
+        throw winrt::hresult_error(RO_E_CLOSED, L"Cannot create CoreWebView2 (via Source) for Closed WebView2 element.");
+    }
+    else
+    {
+        m_creationInProgressAsync = std::make_unique<AsyncWebViewOperations>();
+        RegisterXamlHandlers();
+
+        // We are about to attempt re-creation of the environment,
+        // so clear any previous 'Missing Anaheim Warning'
+        Content(nullptr);
+
+        if (!m_isCoreFailure_BrowserExited_State)
+        {
+            // Normally we always need a new environment, the exception being when Anaheim process failed and we get to reuse the existing one
+            co_await CreateCoreEnvironment();
+        }
+
+        if (m_coreWebViewEnvironment)
+        {
+            co_await CreateCoreWebViewFromEnvironment(GetHostHwnd());
+
+            // Do initialization including rendering setup.
+            // Try this now but defer to Loaded event if it hasn't fired yet.
+            TryCompleteInitialization();
+        }
+        else if (m_shouldShowMissingAnaheimWarning)
+        {
+            CreateMissingAnaheimWarning();
+        }
+    }
+
+   co_return;
+}
+
+winrt::IAsyncAction WebView2::CreateCoreEnvironment() noexcept
+{
+    hstring browserInstall;
+    hstring userDataFolder;
+
+    if (!m_options)
+    {
+        // NOTE: To enable Anaheim logging, use: L"--no-sandbox --enable-features=msEmbeddedBrowserVisualHosting --enable-logging=stderr --v=1"
+        m_options = winrt::CoreWebView2EnvironmentOptions();
+        m_options.AdditionalBrowserArguments(L"--enable-features=msEmbeddedBrowserVisualHosting");
+
+        auto applicationLanguagesList = winrt::ApplicationLanguages::Languages();
+        if (applicationLanguagesList.Size() > 0)
+        {
+            m_options.Language(applicationLanguagesList.GetAt(0));
+        }
+    }
+
+    try
+    {
+        m_coreWebViewEnvironment = co_await winrt::CoreWebView2Environment::CreateWithOptionsAsync(browserInstall, userDataFolder, m_options);
+    }
+    catch (winrt::hresult_error e)
+    {
+        winrt::hresult hr = e.code().value;
+        m_shouldShowMissingAnaheimWarning = hr == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
+        FireCoreWebView2Initialized(hr);
```

Comment:
shouldn't this fire for the success case as well (with S_OK or something like that?)